Testscripts/Linux/Linux-Kernel-Self-Tests.sh

```diff
@@ -0,0 +1,322 @@
+#    5. Collects results
+# 
+# 1)If we set LKS_VERSION_GIT_TAG value, we use Kselftests from stable kernel of
+#   this tag for avoiding unstable builds or non-uniform output format. 
+#   For Centos/Redhat or Ubuntu 16.04, there are maybe some subsystems compiled failed
+#   for that some dependency packages' version is low. We use SKIP_TARGETS_FOR_XXX to
+#   skip these subsystems 
+# 2)If LKS_VERSION_GIT_TAG is NULL, we use Kselftests from distro's own kernel. Maybe
+#   some subsystems are unstable or compiled failed.
+# 3)If LKS_VERSION_GIT_TAG is NUL and this test runs against custom kernel, we use
+#   Kselftests from linux-stable or linux-next of the same version.
+# 4)We just support Ubuntu/Debian/Centos/Redhat now, we will support other distros in the future.
+#
+###############################################################################
+LKS_SRCDIR=""
+MAKEFILE="./tools/testing/selftests/Makefile"
+SKIP_TARGETS="android breakpoints cpu-hotplug memory-hotplug vm"
+SKIP_TARGETS_FOR_CENTOS="bpf intel_pstate networking/timestamping proc size x86"
+SKIP_TARGETS_FOR_UBUNTU16="bpf drivers/dma-buf filesystems/binderfs proc"
+TOTAL_TARGETS=""
+
+ubuntu_src_git="git://git.launchpad.net/~canonical-kernel/ubuntu/+source/linux-azure/+git/"
+next_kernel_src="https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git"
+stable_kernel_src="git://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git"
+
+###############################################################################
+function install_dependencies() {
+	LogMsg "Installing dependencies packages"
+
+	GetDistro
+	update_repos
+
+	case $DISTRO in
+		ubuntu* | debian*)
+			CheckInstallLockUbuntu
+			deb_packages=(make git gcc flex bison clang llvm fuse gcc-multilib libfuse2 \
+						libc6-i386 libc6-dev-i386 libelf-dev libcap-ng-dev libfuse-dev \
+						libpopt-dev libnuma-dev libmount-dev libcap-dev build-essential \
+						pkg-config bc rsync)
+			install_package "${deb_packages[@]}" >> $BUILDING_LOG 2>&1
+			;;
+		centos_7 | centos_8 | redhat_7 | redhat_8)
+			rpm_packages=(make git gcc flex bison clang llvm fuse libcap-ng-devel popt-devel \
+						libmount-devel libcap-devel glibc-devel.*i686 fuse-devel elfutils-devel \
+						numactl-devel glibc-devel)
+			install_epel
+			install_package "${rpm_packages[@]}" >> $BUILDING_LOG 2>&1
+			;;
+		*)
+			LogMsg "Unsupported distro: $DISTRO"
+			SetTestStateSkipped
+			exit 0
+			;;
+	esac
+}
+
+# If the kernel is distro kernel, we download distro kernel source code.
+# if the kernel is custom kernel, we download next or stable kernel source code.
+function download_lks_src() {
+	kernel_version="$(uname -r)"
+
+	# The kernel is custom kernel
+	if [[ $DISTRO =~ "ubuntu" && $kernel_version != *azure* 
+		|| $DISTRO =~ "debian" && $kernel_version != *cloud* 
+		|| $DISTRO =~ "centos" && $kernel_version != *el* 
+		|| $DISTRO =~ "redhat" && $kernel_version != *el* ]]; then
+		if [[ $kernel_version =~ "next" ]]; then
```

Comment:
This if-statement true/false blocks look similar. I think this is high code-duplication. Can you improve?

Reply:
Maybe you want to keep as-is

Reply:
Yeah, sorry I have no idea how I can improve.